test/Microsoft.AspNetCore.Routing.Tests/Template/RoutePrecedenceTestsBase.cs

```diff
@@ -2,13 +2,11 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
-using Microsoft.Extensions.Options;
-using Moq;
 using Xunit;
 
 namespace Microsoft.AspNetCore.Routing.Template
 {
-    public class RoutePrecedenceTests
+    public abstract class RoutePrecedenceTestsBase
```

Comment:
Tests in abstract class?

Reply:
They are run in classes that inherit from it